viite-backend/viite-main/src/main/scala/fi/liikennevirasto/viite/process/strategy/TerminatedTwoTrackSectionSynchronizer.scala

```diff
@@ -0,0 +1,462 @@
+
+    val processedLinks = {
+      var updatedProjectLinks = projectLinks
+      if (minorDiscontinuitiesToProcess.nonEmpty) {
+        // Process Minor discontinuity link pairs one by one
+        minorDiscontinuitiesToProcess.foreach({ minorDiscontinuityLinks =>
+          // Find terminated sections that are located right after the minor discontinuity link(s)
+          val (leftTerminatedAfterMinorDisc, rightTerminatedAfterMinorDisc) = minorDiscontinuityLinks match {
+            case Seq(combined) =>
+              val leftTerminatedAfterMinorDisc = updatedProjectLinks.find(pl => pl.status == RoadAddressChangeType.Termination && pl.track == Track.LeftSide && combined.originalAddrMRange.continuesTo(pl.originalAddrMRange))
+              val rightTerminatedAfterMinorDisc = updatedProjectLinks.find(pl => pl.status == RoadAddressChangeType.Termination && pl.track == Track.RightSide && combined.originalAddrMRange.continuesTo(pl.originalAddrMRange))
+              (leftTerminatedAfterMinorDisc, rightTerminatedAfterMinorDisc)
+            case Seq(left, right) =>
+              val leftTerminatedAfterMinorDisc = updatedProjectLinks.find(pl => pl.status == RoadAddressChangeType.Termination && pl.track == Track.LeftSide && left.originalAddrMRange.continuesTo(pl.originalAddrMRange))
+              val rightTerminatedAfterMinorDisc = updatedProjectLinks.find(pl => pl.status == RoadAddressChangeType.Termination && pl.track == Track.RightSide && right.originalAddrMRange.continuesTo(pl.originalAddrMRange))
+              (leftTerminatedAfterMinorDisc, rightTerminatedAfterMinorDisc)
+            case _ => (None,None)
+          }
+
+          if (leftTerminatedAfterMinorDisc.isDefined && rightTerminatedAfterMinorDisc.isDefined) {
+            // Update terminated links to match, the discontinuity links to match and if there are links after terminated section, adjust those as well
+            // Get the terminated sections for both tracks
+            val leftTermSect  = terminatedLeftSections.find( section => section.exists(_.id == leftTerminatedAfterMinorDisc.get.id))
+            val rightTermSect = terminatedRightSections.find(section => section.exists(_.id == rightTerminatedAfterMinorDisc.get.id))
+
+            val firstTerminatedLeft   = leftTermSect.get.minBy(_.addrMRange.start)
+            val firstTerminatedRight  = rightTermSect.get.minBy(_.addrMRange.start)
+
+            val lastTerminatedLeft  = leftTermSect.get.maxBy(_.addrMRange.end)
+            val lastTerminatedRight = rightTermSect.get.maxBy(_.addrMRange.end)
+
+            val averageStartForTermSect = calculateAverageAddrM(firstTerminatedLeft.addrMRange.start, firstTerminatedRight.addrMRange.start)
+            val averageEndForTermSect   = calculateAverageAddrM(lastTerminatedLeft.addrMRange.end, lastTerminatedRight.addrMRange.end)
+
+            val updatedMinorDiscLinks = {
```

Comment:
Tän vois ehkä vastaavasti tiputtaa tonne alemmas, missä tätä käytetään, niin on säännönmukaisesti käyttö ja esittely vierekkäin, kuten ton adjustTerminatedLinks kohdallakin